sections/git.zsh

```diff
@@ -10,31 +13,32 @@ SPACESHIP_GIT_SHOW="${SPACESHIP_GIT_SHOW=true}"
 SPACESHIP_GIT_PREFIX="${SPACESHIP_GIT_PREFIX="on "}"
 SPACESHIP_GIT_SUFFIX="${SPACESHIP_GIT_SUFFIX="$SPACESHIP_PROMPT_DEFAULT_SUFFIX"}"
 SPACESHIP_GIT_SYMBOL="${SPACESHIP_GIT_SYMBOL=" "}"
-
-# ------------------------------------------------------------------------------
-# Dependencies
-# ------------------------------------------------------------------------------
-
-source "$SPACESHIP_ROOT/sections/git_branch.zsh"
-source "$SPACESHIP_ROOT/sections/git_status.zsh"
+if [ -z "$SPACESHIP_GIT_ORDER" ]; then
+  SPACESHIP_GIT_ORDER=(
+    branch
+    status
+  )
+fi
 
 # ------------------------------------------------------------------------------
 # Section
 # ------------------------------------------------------------------------------
 
-# Show both git branch and git status:
-#   spaceship_git_branch
-#   spaceship_git_status
 spaceship_git() {
-  [[ $SPACESHIP_GIT_SHOW == false ]] && return
+  [[ $SPACESHIP_GIT_SHOW == false || -z $SPACESHIP_GIT_ORDER ]] && return
+
+  spaceship::is_git || return
+
+  local git_prompt
 
-  local git_branch="$(spaceship_git_branch)" git_status="$(spaceship_git_status)"
+  for el in ${(Oa)SPACESHIP_GIT_ORDER}; do
+    source "$SPACESHIP_ROOT/sections/git_${el}.zsh" # load dependency
```

Comment:
this will source every time a prompt is refreshed, in contrast the rest of the sections are sourced only once - I'd suggest moving the "source" part to `spaceship.zsh` to the same place where other sections are sourced.

Reply:
Good catch thanks. Addressed in 2b91ca2. I'm just sourcing all git sections since that's what Spaceship does for everything else. (Maybe not ideal, but that can be discussed in #407)